### PR TITLE
 Update subscribing to PeerManagerNotfications to take a sender 

### PR DESCRIPTION
**Pull request by agunde406**
This commit deprecates subscribe() in favor of subscribe_sender
which takes a Sender<T> which can be used to get a
PeerManagerNotfication and adds the ability to convert the
notification to another type. This change updates the PeerManager
to match the ConnectionManager.

Also adds an unsubscribe option to PeerManagerConnector so a
subscriber can unsubscribe from notifications.

Also removes Clone and PartialEq from PeerManager Messages.
These decorators were not required and caused issues with the
adding of the new subscriber messages.

### CODE REVIEW

File: libsplinter/src/peer/connector.rs [hunk 1]
```diff
@@ -203,6 +204,66 @@ impl PeerManagerConnector {
...
     }
+
+    /// Subscribe to notifications for peer events.
+    ///
+    /// `PeerManagerNotification` instances will be transformed via type `T`'s implementation
+    /// of `From<PeerManagerNotification>` and passed to the given sender.
+    ///
+    /// # Returns
+    ///
+    /// The subscriber id that can be used for unsubscribing the given sender.
```
Comment (ltseeley) [inline libsplinter/src/peer/connector.rs:215]: s/id/ID/

File: libsplinter/src/peer/connector.rs [hunk 2]
```diff
@@ -203,6 +204,66 @@ impl PeerManagerConnector {
...
+            .map_err(|_| {
+                PeerManagerError::SendMessageError("The peer manager is no longer running".into())
+            })?;
+
+        recv.recv().map_err(|_| {
+            PeerManagerError::SendMessageError("The peer manager is no longer running".into())
+        })?
+    }
+
+    /// Unsubscribe to `PeerManagerNotification`.
```
Comment (ltseeley) [inline libsplinter/src/peer/connector.rs:245]: s/to/from/

File: libsplinter/src/peer/notification.rs [hunk 3]
```diff
@@ -61,6 +63,51 @@ impl Iterator for PeerNotificationIter {
...
+        let mut failures = vec![];
+        for (id, callback) in self.subscribers.iter() {
+            if let Err(err) = (*callback)(notification.clone()) {
+                failures.push(*id);
+                debug!("Dropping subscriber ({}): {}", id, err);
+            }
+        }
+
+        for id in failures {
+            self.subscribers.remove(&id);
```
Comment (ltseeley) [inline libsplinter/src/peer/notification.rs:94]: Why can't this be done in the `if let Err...`?

File: libsplinter/src/peer/connector.rs (hunk 1 above)
Comment (agunde406) [reply libsplinter/src/peer/connector.rs]: fixed

File: libsplinter/src/peer/connector.rs (hunk 2 above)
Comment (agunde406) [reply libsplinter/src/peer/connector.rs]: fixed

File: libsplinter/src/peer/notification.rs (hunk 3 above)
Comment (agunde406) [reply libsplinter/src/peer/notification.rs:94]: An immutable borrow occurs above

**Pull request merged**